src/eredis_client.erl

```diff
@@ -528,18 +533,24 @@ do_sentinel_reconnect(Host, Port, State) ->
 -spec start_reconnect(#state{}) -> {ok, #state{}}.
 start_reconnect(#state{socket=undefined} = State) ->
     Self = self(),
-    spawn(fun() -> reconnect_loop(Self, State) end),
+    Options = get_options(),
+    spawn_link(fun() -> reconnect_loop(Self, State, Options) end),
```

Comment:
this will contaminate the parent process' mailbox with EXIT message.
I would recommend to call `spawn(?MODULE, reconnect_loop, [Self, State, Options])`
and in the connect_loop, it checks `Self`'s aliveness the first thing entering the loop.

Reply:
Or keep track of this pid, and handle it's `EXIT` message properly.
However that would require a change in the `#state` record which complicates the upgrade/downgrade logic.

Reply:
Not Really. 
If we do that, the main process can't monitor the child's abnormal exit. 
we link them together to make sure child abnormal exit will make main process exit too.